**Data Science and Machine Learning:  in Protein-Protein Network Analysis**

**Authors: Rachael Oluwakamiye Abolade (@Oluwakamiye)**

_Proteins_ are organic molecules abundant in living systems that conducts a range of unique functions such as transport, storage, membrane composition, and enzymatic action, among others (Clark, 2018). Thus, one may obtain biological insights into protein functions, disease prevalence, and therapy development by identifying interaction amongst protein pairs (Zhao _et al_., 2020).****

Protein–protein interactions compose complexes to conduct numerous biological processes and functions such as metabolic cycles, signal transduction, DNA transcription and replication, catalysis, and immune response (Jones _et al_., 1996). The activities of cells and their functions are affected by abnormalities in protein interactions, leading to numerous diseases such as cancer. Comprehensive identification of PPIs can help to decode the molecular mechanisms of the specific biological functions involved (Wang _et al_., 2020). The proximity of proteins in PPI is of paramount importance for specific functionality. Jansen et al. (2003) demonstrated that the interaction between known and unknown functional proteins can significantly contribute toward deciphering many protein functions.

PPIs may help in decoding the functionality of unannotated proteins. Many experimental studies have been conducted to identify PPI, such as the yeast two–hybrid among others (Noor _et al_., 2021). However, these approaches are laborious and time-consuming, which makes them difficult to employ for all protein pairs (Zhou _et al_., 2016). Therefore, one may use computational methods as pre-treatment in advance of the experimental methods, to reduce false-positive and false-negative results (Wang _et al_., 2018).__

 

While biological descriptions seem to have reached levels of information, powerful computational approaches became available for data science. They are based on machine learning (ML), a procedure that can discover relations, if existing, among dependent and independent data sets (Baldi, 2021).

ML refers to many algorithms able to automatically build models for inference and clustering, starting from a set of data called _training set_. ML can be unsupervised or supervised. In the context of computing PPIs, supervised ML is most relevant

Supervised ML algorithms infer mapping between two spaces (input and output), based on known training examples (Bishop, 2006). Supervised ML aims to implement tools that generalize the learned associations to new examples. Supervised ML methods can be adopted for classification or regression, depending on the discrete or continuous property of the output space.

Recently, some improvements in networks validation were described, including an analysis of network paths (Kovacs _et al_., 2019) and its variant integrating complementary interface and gene duplication (Chen _et al_., 2020).

 

**References**

Baldi P. _Deep learning in science_. Cambridge: Cambridge University Press; 2021;371.

Bishop C.M. _Pattern recognition and machine learning. Information science and statistics_. New York: Springer; 2006;738

Chen Y, Wang W, Liu J, Feng J, Gong X. Protein Interface complementarity and gene duplication improve link prediction of protein-protein interaction network. _Front Genet_. 2020; **11**: 291.

 Jansen R., Yu H., Greenbaum D., Kluger Y., Krogan N.J., Chung S., Emili A., Snyder M., Greenblatt J.F., Gerstein M. bayesian networks approach for predicting protein-protein interactions from genomic data. _Science. _2003; 302:449–453

Jones S., Thornton J.M. Principles of protein-protein interactions. _Proceedings of the National Academy of Sciences. _1996;93(1):13–20

Kovács I.A, Luck K, Spirohn K, Wang Y, Pollis C, Schlabach S, et al. Network-based prediction of protein interactions. _Nat Commun_. 2019; **10**(1): 1240.

M.A. Clark, J. Choi, Biology (2018).

Mackay J.P., Sunde M., Lowry J.A., Crossley M., Matthews J.M. Protein interactions: is seeing believing? _Trends in Biochemical Sciences. _2007;32(12):530–531.

Wang T., Li L., Huang Y.-A., Zhang H., Ma Y., Zhou X. Prediction of protein-protein interactions from amino acid sequences based on continuous and discrete wavelet transform features. _Molecules. _2018;23(4):823. 

Wang Y., You Z., Li L., Chen Z. A survey of current trends in computational predictions of protein-protein interactions. _Frontiers of Computer Science. _2020;14(4):1–12. 

Zhang F., Shi W., Zhang J., Zeng M., Li M., Kurgan L. Probselect: accurate prediction of protein-binding residues from proteins sequences via dynamic predictor selection. _Bioinformatics. _2020;36: i735–i744.

Zhao J., Cao Y., Zhang L. Exploring the computational methods for protein-ligand binding site prediction. _Computational and structural biotechnology journal. _2020; 18:417–426.

Zhou M., Li Q., Wang R. Current experimental methods for characterizing protein–protein interactions. _ChemMedChem. _2016;11(8):738.
